From b82bf51cfe0ebf74e1f5ae22b9aafed3aee1fa66 Mon Sep 17 00:00:00 2001
From: Minh Le <minh.le.aj@renesas.com>
Date: Sat, 10 May 2025 18:35:30 +0700
Subject: [PATCH 24/52] arm64: renesas: gen5-common: Allow WDT reset

R-Car X5H firmware misses to enable WDT reset. Enable that in U-Boot.

Signed-off-by: Minh Le <minh.le.aj@renesas.com>
Signed-off-by: Hai Pham <hai.pham.ud@renesas.com>
---
 board/renesas/common/gen5-common.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/board/renesas/common/gen5-common.c b/board/renesas/common/gen5-common.c
index ea1725a1c56..7dad9588f9b 100644
--- a/board/renesas/common/gen5-common.c
+++ b/board/renesas/common/gen5-common.c
@@ -54,6 +54,10 @@ int board_early_init_f(void)
 
 int board_init(void)
 {
+	/* Allow WDT reset */
+	writel(RST_KCPROT_DIS, RST_RESKCPROT0);
+	clrbits_le32(RST_WDTRSTCR, RST_WWDT_RSTMSK | RST_RWDT_RSTMSK);
+
 	if (current_el() != 3)
 		return 0;
 	init_gic_v3();
-- 
2.34.1

